testing/matrix-sdk-integration-testing/src/tests/sliding_sync/room.rs

```diff
@@ -538,3 +550,299 @@ async fn test_room_notification_count() -> Result<()> {
 
     Ok(())
 }
+
+static DROP_TODEVICE: StdMutex<bool> = StdMutex::new(true);
+fn drop_todevice(response: &mut Bytes) {
+    let Ok(mut json) = serde_json::from_slice::<serde_json::Value>(response) else {
+        return;
+    };
+    let Some(extensions) = json.get_mut("extensions").and_then(|e| e.as_object_mut()) else {
+        return;
+    };
+    let Some(to_device) = extensions.remove("to_device") else {
+        return;
+    };
```

Comment:
Any chance we can avoid all the short-circuiting by using `map` and other transformer functions like that? Otherwise, let's comment why it's ok to return in each case.

Reply:
I added some comments, let me know if this needs more changes.